source/common/http/http2/codec_impl.h

```diff
@@ -381,6 +383,11 @@ class ConnectionImpl : public virtual Connection, protected Logger::Loggable<Log
 
     RequestDecoder* request_decoder_{};
     absl::variant<RequestHeaderMapPtr, RequestTrailerMapPtr> headers_or_trailers_;
+    absl::optional<bool> stream_error_on_invalid_http_message_;
+
+    absl::optional<bool> streamErrorOnInvalidHttpMessage() const override {
+      return stream_error_on_invalid_http_message_;
```

Comment:
Do we need to latch this, or can we just return parent_.stream_error_on_invalid_http_messaging_ here?

Also let's make sure this and the HTTP/1.1 function have unit tests, even if they're pretty trivial.